subprojects/logging/src/main/java/org/gradle/internal/logging/progress/DefaultProgressLoggerFactory.java

```diff
@@ -145,12 +142,23 @@ public void started(String status) {
                 parent.assertRunning();
             }
             current.set(this);
-            listener.started(new ProgressStartEvent(progressOperationId, parent == null ? null : parent.progressOperationId, timeProvider.getCurrentTime(), category, description, shortDescription, loggingHeader, toStatus(status), buildOperationId));
+            ProgressStartEvent progressStartEvent = new ProgressStartEvent(
+                progressOperationId,
+                parent == null ? null : parent.progressOperationId,
+                timeProvider.getCurrentTime(),
+                category,
+                description,
+                shortDescription,
+                loggingHeader,
+                toStatus(status),
+                getBuildOperationId(),
+                buildOperationDescriptor == null ? null : buildOperationDescriptor.getOperationType());
```

Comment:
could default to 'uncategorized' so that the property is not nullable on the event, and so we don't need to write a 'null' flag when serializing (along with all the other good reasons why not-nullable would be good).